### PR TITLE
Calendar revamp with events

### PR DESCRIPTION
**Pull request by justagoodguy-omega**

### CODE REVIEW

File: app/src/androidTest/java/com/github/onedirection/navigation/CalendarUITest.java [hunk 1]
```diff
@@ -93,30 +91,103 @@ public void changingMonthsCorrectlyUpdatesCalendarTest() {
...
+                                        0),
+                                10)));
+        materialButton2.perform(scrollTo(), click());
+        Thread.sleep(20000);
+        DataInteraction linearLayout = onData(anything())
+                .inAdapterView(allOf(withId(R.id.gridView),
+                        childAtPosition(
+                                withClassName(is("android.widget.LinearLayout")),
+                                2)))
+                .atPosition(15);
```
Comment (TheImpressiveDonut) [inline app/src/androidTest/java/com/github/onedirection/navigation/CalendarUITest.java:191]: I think this doesn't test anything

File: app/src/androidTest/java/com/github/onedirection/navigation/TEMP.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.anything;
+import static org.hamcrest.Matchers.is;
+
+@LargeTest
+@RunWith(AndroidJUnit4.class)
+public class TEMP {
+
```
Comment (TheImpressiveDonut) [inline app/src/androidTest/java/com/github/onedirection/navigation/TEMP.java:41]: I suppose the class name means it is temporary, shouldn't you delete it or maybe you want to keep it for later

File: app/src/main/java/com/github/onedirection/navigation/fragment/calendar/CalendarGridAdapter.java [hunk 3]
```diff
@@ -51,14 +55,24 @@ public View getView(int position, @Nullable View convertView, @NonNull ViewGroup
...
             for (int i = 0; i < events.size(); ++i) {
-
+                eventCalendar.setTime(Date.from(Instant.ofEpochSecond(events.get(i).getStartTime().toEpochSecond())));
+                if (dayNumber == eventCalendar.get(Calendar.DAY_OF_MONTH) && displayMonth == eventCalendar.get(Calendar.MONTH) + 1
+                        && displayYear == eventCalendar.get(Calendar.YEAR)) {
+                    nbOfEventsInDay++;
+                    if (nbOfEventsInDay == 1) {
+                        EventNumber.setText(nbOfEventsInDay + " Event");
+                    } else {
+                        EventNumber.setText(nbOfEventsInDay + " Events");
```
Comment (TheImpressiveDonut) [inline app/src/main/java/com/github/onedirection/navigation/fragment/calendar/CalendarGridAdapter.java:73]: There are some lints warning here

File: app/src/main/res/layout/day_select_popup.xml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:orientation="vertical">
+    <Button
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="Add Event"
```
Comment (TheImpressiveDonut) [inline app/src/main/res/layout/day_select_popup.xml:9]: There is one lint warning here.

File: app/src/main/res/layout/day_select_popup.xml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        android:layout_gravity="center"
+        android:layout_marginHorizontal="10dp"
+        android:textAlignment="center"
+        android:textColor="@color/black"
+        android:id="@+id/addEvent" />
+
+    <Button
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="View Events"
```
Comment (TheImpressiveDonut) [inline app/src/main/res/layout/day_select_popup.xml:19]: There is one lint warning here.

File: app/src/main/res/layout/custom_calendar_view.xml [hunk 6]
```diff
@@ -6,7 +6,6 @@
     android:layout_width="match_parent"
     android:layout_height="match_parent"
     android:orientation="vertical"
-    android:id="@+id/activity_custom_calendar"
     tools:context=".navigation.fragment.calendar.CustomCalendarView">
 
     <LinearLayout
```
Comment (TheImpressiveDonut) [inline app/src/main/res/layout/custom_calendar_view.xml:11]: Would be nice if you could fix the lint warning in this file too if it can be done

**Review by TheImpressiveDonut — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/github/onedirection/navigation/fragment/calendar/CustomCalendarView.java [hunk 7]
```diff
@@ -89,13 +131,51 @@ private void SetUpCalendar() {
...
+    }
+
+    private void callEventCreator(int year, int month, int day) {
+        Intent intent = new Intent(parentFragment.getActivity(), EventCreator.class);
+        EventCreator.putDateExtra(intent, LocalDate.of(year, month, day));
+        parentFragment.startActivity(intent);
+    }
+
+    private void callDayEventsList() {
+        //@TODO when the events list is implemented;
```
Comment (Sosol02) [inline app/src/main/java/com/github/onedirection/navigation/fragment/calendar/CustomCalendarView.java:175]: Consider throwing an UnsupportedOperationException to catch bugs if someone calls this function expecting it to be implemented.

**Review by Sosol02 — CHANGES_REQUESTED** [review]

File: app/src/androidTest/java/com/github/onedirection/navigation/CalendarUITest.java [hunk 8]
```diff
@@ -93,30 +150,49 @@ public void changingMonthsCorrectlyUpdatesCalendarTest() {
                                 2),
                         isDisplayed()));
         navigationMenuItemView.perform(click());
-        ViewInteraction textView = onView(
-                allOf(withId(R.id.currentDate),
-                        withParent(withParent(withId(R.id.activity_custom_calendar))),
-                        isDisplayed()));
-        textView.check(matches(isDisplayed()));
-        firstDate = textView.toString();
+        Thread.sleep(20000);
```
Comment (hugolan) [inline app/src/androidTest/java/com/github/onedirection/navigation/CalendarUITest.java:153]: do you need the Thread.sleep(20000)?

File: app/src/main/res/layout/custom_calendar_view.xml [hunk 9]
```diff
@@ -44,6 +44,7 @@
             android:id = "@+id/nextBtn"
             android:background="@mipmap/shrek_dab_right"
             android:layout_margin = "10dp"
+            android:contentDescription="Our lord and savior dabbing to the right"
```
Comment (hugolan) [inline app/src/main/res/layout/custom_calendar_view.xml:47]: plz stop

File: app/src/main/res/layout/custom_calendar_view.xml [hunk 10]
```diff
@@ -23,10 +22,11 @@
             android:id = "@+id/previousBtn"
             android:background="@mipmap/shrek_dab_left"
             android:layout_margin = "10dp"
+            android:contentDescription="Our lord and savior dabbing to the left"
```
Comment (TheImpressiveDonut) [inline app/src/main/res/layout/custom_calendar_view.xml:25]: There is a lint warning

File: app/src/main/res/layout/custom_calendar_view.xml (hunk 9 above)
Comment (TheImpressiveDonut) [inline app/src/main/res/layout/custom_calendar_view.xml:47]: There is a lint warning

**Review by TheImpressiveDonut — CHANGES_REQUESTED** [review]
Lint warnings !

**Review by TheImpressiveDonut — APPROVED** [review]
Nice !

**Review by Sosol02 — APPROVED** [review]

**Pull request merged**